packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -363,6 +367,62 @@ async function processBlogSourceFile(
   };
 }
 
+const sortPresets: SortPresets = {
+  descending: ({blogPosts}) =>
+    blogPosts.sort(
+      (a, b) => b.metadata.date.getTime() - a.metadata.date.getTime(),
+    ),
+  ascending: ({blogPosts}) =>
+    blogPosts.sort(
+      (a, b) => a.metadata.date.getTime() - b.metadata.date.getTime(),
+    ),
+};
+
+interface SortBlogPostsOptions {
+  blogPosts: BlogPost[];
+  sortPosts: SortBlogPostsPreset | SortBlogPostsFn;
+}
+
+function getSortFunction(sortPosts: Options['sortPosts']): SortBlogPostsFn {
+  if (typeof sortPosts === 'function') {
+    return sortPosts;
+  }
+
+  if (typeof sortPosts === 'string') {
+    const presetFn = sortPresets[sortPosts];
+    if (!presetFn) {
+      throw new Error(
+        `sortPosts preset ${sortPosts} does not exist, valid presets are: ${Object.keys(
+          sortPresets,
+        ).join(', ')}`,
+      );
+    }
+    return presetFn;
+  }
+
+  return () => {};
+}
+
+export function sortBlogPosts({
+  blogPosts,
+  sortPosts,
+}: SortBlogPostsOptions): BlogPost[] {
+  const sortFunction = getSortFunction(sortPosts);
+  const sortedBlogPosts = sortFunction({blogPosts});
+
+  if (sortedBlogPosts !== undefined) {
+    if (sortedBlogPosts.length === 0) {
```

Comment:
I'm not sure why someone would return an empty array by mistakes.

Users are likely to either return undefined, or return a new sorted list of posts. 

If we want to prevent mistakes, we should check that `blogPosts.length === sortedBlogPosts.length`.

We could check that the returned value is an array, because users might use JS in config files and return an unsafe value such as a number 🤷‍♂️ 

---

But I'm not sure we want to prevent that anyway, because `sortPosts` could also be used as a temporary workaround to filter/modify posts until we come up with a better API for it.
It's not the primary use case of this API, and it's a bit awkward to do so.
See my comment here:
https://github.com/facebook/docusaurus/pull/9846#discussion_r1490916996

Maybe `sortPosts` is not the API we should extend?